server.cjs

```diff
@@ -0,0 +1,51 @@
+const { createServer, ServerResponse, IncomingMessage } = require('http')
+const { parse } = require('url')
+const next = require('next')
+ 
+const dev = process.env.NODE_ENV !== 'production'
+const hostname = 'localhost'
+const port = 3000
+// when using middleware `hostname` and `port` must be provided below
+const app = next({ dev, hostname, port })
+const handle = app.getRequestHandler()
+
+app.prepare().then(() => {
+  createServer(async (req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+
+    const setHeader = res.setHeader
+    let cacheControlOverwritten = false
+    res.setHeader = function(header, value) {
+      if (header.toLowerCase() === 'x-cache-control') {
```

Comment:
And here can we use const from config?

Reply:
It's not an easy thing to import something both to cjs and mjs, but I will check if server can be mjs

Reply:
added